src/utils/fetchUtil.js

```diff
@@ -15,26 +20,33 @@ export default class fetchUtil {
       var contentType = 'application/json';
       if (this.useActionParam) {
         url = url + '?action=' + actionId;
-      }
-      else {
+      } else {
         contentType = 'application/vnd.pingidentity.' + actionId + '+json';
       }
       headers['Content-Type'] = contentType;
     }
+    // add more headers
+    httpHeaders.forEach((value, key) => {
+      headers[key] = value;
+    });
+    //options
     var options = {
       headers: headers,
       method: method,
       body: body,
-      credentials: 'include'
+    }
+    // include credentials
+    if (!this.cookieless) {
+      options.credentials = 'include'
     }
     return fetch(url, options);
   }
 
-  getFlow(flowId) {
-    return this.doRequest('GET', flowId);
+  getFlow(flowId, headers = new Map()) {
+    return this.doRequest('GET', flowId, headers);
```

Comment:
The `actionId` and the `body` param are missing here if you want to call `doRequest` with the `headers`. Perhaps the `headers` can be defaulted in `doRequest` instead?

Reply:
good catch